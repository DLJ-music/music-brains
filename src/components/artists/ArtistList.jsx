import React from 'react';
import Artist from './Artist.jsx';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ArtistList = ({ artists }) => {
  const artistElements = artists.map(artist => (
    <Link key={artist.id} to={`/artist/${artist.id}`} >
      <li key={artist.id}>
        <Artist {...artist}/>
      </li>
    </Link>
  ));
  return (
    <ul>
      {artistElements}
    </ul>
  );
};

ArtistList.propTypes = {
  artists: PropTypes.array.isRequired
};

export default ArtistList;

