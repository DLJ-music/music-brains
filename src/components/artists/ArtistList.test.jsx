import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from './ArtistList';

describe('ArtistList component', () => {
  let artistListInfo;
  beforeEach(() => {
    artistListInfo = [{ id: 'stjkstkjlljst', name: 'Jackson Michael', type: 'Group', score: 87 }];
  });

  it('renders ArtistList', () => {
    const wrapper = shallow(<ArtistList artists={artistListInfo}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
