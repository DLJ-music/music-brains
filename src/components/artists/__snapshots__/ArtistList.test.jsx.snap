// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders ArtistList 1`] = `
<ul>
  <Link
    key="stjkstkjlljst"
    to="/artist/stjkstkjlljst/Jackson Michael"
  >
    <li
      key="stjkstkjlljst"
    >
      <Artist
        id="stjkstkjlljst"
        name="Jackson Michael"
        score={87}
        type="Group"
      />
    </li>
  </Link>
</ul>
`;
